### PR TITLE
Add error for using `set_index` in read_csv

### LINKED ISSUE #714: read_csv: passing index_col results in KeyError
Sample data:
```
"id1","id2","attr1","attr2","attr3","attr4"
1072.0,"2014-02-22 12:30:00",1,93440248,"False",750
1072.0,"2014-02-22 12:45:00",1,93440248,"False",765
1072.0,"2014-02-22 13:00:00",1,93440248,"False",780
1072.0,"2014-02-22 13:15:00",1,93440248,"False",795
```

The following results in a `KeyError` (full stack trace on [pastebin.com](http://pastebin.com/nx1zBQA7)):
```python
dask_df = dd.read_csv('test.csv',  index_col='id1')
```

However, setting the index column using the `set_index` method works (e.g. as follows) for single-columned indexes ([MultiIndexes](http://pandas.pydata.org/pandas-docs/stable/advanced.html) are not yet supported in Dask).

```python
dask_df = dd.read_csv('test.csv')
dask_df = dask_df.set_index('id1')
```

Issue detected when using Dask 0.7.1 and pandas 0.16.2.

### PR DESCRIPTION
**Pull request by cowlicks**
closes #714 

I didn't add a test, but verified that this works locally. I can add one, but I wasn't sure if this needed one.

### CODE REVIEW

File: dask/dataframe/io.py [hunk 1]
```diff
@@ -91,6 +91,14 @@ def fill_kwargs(fn, args, kwargs):
     kwargs: dict
         keyword arguments to give to pd.read_csv
     """
+
+    if 'index_col' in kwargs:
+        msg = """
+        The index column cannot be set at dataframe creation time. Instead use
+        the `set_index` method on the dataframe after it is created.
+        """
+        raise ValueError(msg)
```
Comment (mrocklin) [inline dask/dataframe/io.py]: Perhaps the following pattern

```python
def test_set_index():
    ...
    try:
        dd.read_csv(..., index_col='foo')
        assert False
    except ValueError as e:
        assert 'set_index' in str(e)
```

Comment (mrocklin) [inline dask/dataframe/io.py]: ping

Comment (cowlicks) [inline dask/dataframe/io.py]: Test added.

Comment (cowlicks) [inline dask/dataframe/io.py]: @mrocklin pong, test passing

**Pull request merged**